service/history/api/describeworkflow/api.go

```diff
@@ -155,6 +155,16 @@ func Invoke(
 			FirstRunId:                   executionInfo.FirstExecutionRunId,
 			VersioningInfo:               executionInfo.VersioningInfo,
 		},
+		WorkflowExtendedInfo: &workflowpb.WorkflowExecutionExtendedInfo{
+			ExecutionExpirationTime: executionInfo.WorkflowExecutionExpirationTime,
+			RunExpirationTime:       executionInfo.WorkflowRunExpirationTime,
+			OriginalStartTime:       startEvent.EventTime,
+			CancelRequested:         executionInfo.CancelRequested,
+		},
+	}
+
+	if mutableState.IsResetRun() {
+		result.WorkflowExtendedInfo.LastResetTime = executionState.StartTime
```

Comment:
I wonder we should expose OriginalRunID instead instead of this. 
Both can be used to tell if the run is a reset run, but LastResetTime will always be the same as result.WorkflowExecutionInfo.StartTime.

Reply:
Yes, probably you right - by itself LastResetTime doesn't add much value, only the fact that reset happens.
OriginalRunID will add more value - information that reset happen and actual original run id.

good thing - now we have a means to expose whatever we want.